externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/spark/kyuubi/SQLOperationListenerSuite.scala

```diff
@@ -22,16 +22,37 @@ import scala.collection.JavaConverters.asScalaBufferConverter
 import org.apache.hive.service.rpc.thrift.{TExecuteStatementReq, TFetchOrientation, TFetchResultsReq, TOperationHandle}
 import org.scalatest.time.SpanSugar._
 
+import org.apache.kyuubi.config.KyuubiConf
 import org.apache.kyuubi.config.KyuubiConf.OPERATION_SPARK_LISTENER_ENABLED
 import org.apache.kyuubi.engine.spark.WithSparkSQLEngine
 import org.apache.kyuubi.operation.HiveJDBCTestHelper
 
 class SQLOperationListenerSuite extends WithSparkSQLEngine with HiveJDBCTestHelper {
 
-  override def withKyuubiConf: Map[String, String] = Map.empty
+  override def withKyuubiConf: Map[String, String] = Map(
+    KyuubiConf.ENGINE_SPARK_SHOW_PROGRESS.key -> "true",
```

Comment:
They are only needed by the `operation listener with progress job info` test case, we should move them inside the test case and configure them via `withSessionConf`.

Reply:
> They are only needed by the `operation listener with progress job info` test case, we should move them inside the test case and configure them via `withSessionConf`.

get, i will fix it

Reply:
![image](https://github.com/apache/kyuubi/assets/51512358/b1a0a1d8-f2c9-4abe-8d5f-907fcd4ff811)
have set conf in withSessionConf, but it is useless when init SparkConsoleProgressBar,because the conf is not effective, do you know how to solve this problem?
![image](https://github.com/apache/kyuubi/assets/51512358/49d8dd1a-3639-41d4-98b7-78789ba56005)

Reply:
These two configurations in `SQLOperationListener` are not obtained from spark session conf, we can obtain them like this:

```
  private def getSessionConf[T](configEntry: ConfigEntry[T]): T = {
    spark.conf.getOption(configEntry.key).map(configEntry.valueConverter).getOrElse {
      operation.getSession.sessionManager.getConf.get(configEntry)
    }
  }

  private lazy val consoleProgressBar =
    if (getSessionConf(ENGINE_SPARK_SHOW_PROGRESS)) {
      Some(new SparkConsoleProgressBar(
        operation,
        activeStages,
        getSessionConf(ENGINE_SPARK_SHOW_PROGRESS_UPDATE_INTERVAL),
        getSessionConf(ENGINE_SPARK_SHOW_PROGRESS_TIME_FORMAT)))
    } else {
      None
    }
```

cc @cxzl25 WDYT?

Reply:
https://github.com/apache/kyuubi/pull/5410#discussion_r1360162001
if we ignore the config position, just change test case order, the test is also success.

Reply:
Yes, we can improve it in subsequent PRs, please take a look at https://github.com/apache/kyuubi/pull/5410#discussion_r1360160628 and use a faster sql instead of `java_method('java.lang. Thread', 'sleep', 10000l)`, like 

```
select /*+ REPARTITION(3, a) */ a from values(1) t(a);
```

Reply:
i have try the fast sql test locally, but the fast sql looks like will not get the interval make there not have jobInfo log so that test failed, so i use a slow sql to replace, and the local and ci test all success

Reply:
> i have try the fast sql test locally, but the fast sql looks like will not get the interval make there not have jobInfo log so that test failed, so i use a slow sql to replace, and the local and ci test all success

Got it

Reply:
> These two configurations in SQLOperationListener are not obtained from spark session conf, we can obtain them like this:

+1